### PR TITLE
:sparkles: Feature: support multiple config

### PR DESCRIPTION
**Pull request by STDSuperman**
Can do something code and ui review...(#779 )

### CODE REVIEW

**Comment by STDSuperman** [comment]
### UI Preview

File: src/renderer/pages/UploaderConfigPage.vue [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        v-for="item in curConfigList"
+        :key="item._id"
+        :span="11"
+        :offset="1"
+      >
+        <div
+          :class="`config-item ${defaultConfigId === item._id ? 'selected' : ''}`"
+          @click="() => selectItem(item._id)"
+        >
+          <div class="config-name">{{item.name}}</div>
```
Comment (Molunerfinn) [inline src/renderer/pages/UploaderConfigPage.vue:18]: Should provide a way to change the config name

File: src/renderer/pages/UploaderConfigPage.vue [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+          <div class="config-update-time">{{formatTime(item._updatedAt)}}</div>
+          <div class="operation-container">
+            <i class="el-icon-edit" @click="openEditPage(item.type)"></i>
+            <i :class="`el-icon-delete ${curConfigList.length <= 1 ? 'disabled' : ''}`" @click="() => deleteConfig(item._id)"></i>
+          </div>
+        </div>
+      </el-col>
```
Comment (Molunerfinn) [inline src/renderer/pages/UploaderConfigPage.vue:25]: Should add a area for users to add more configs

File: src/renderer/pages/UploaderConfigPage.vue [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      </div>
+      <el-col
+        class="config-item-col"
+        v-for="item in curConfigList"
+        :key="item._id"
+        :span="11"
+        :offset="1"
+      >
+        <div
+          :class="`config-item ${defaultConfigId === item._id ? 'selected' : ''}`"
```
Comment (Molunerfinn) [inline src/renderer/pages/UploaderConfigPage.vue:15]: Just highlighting does not give the users a hint that this is the current selection, it is better to give a clear text hint

**Review by Molunerfinn — CHANGES_REQUESTED** [review]
Thanks for contribution, just a little problems

File: src/renderer/pages/UploaderConfigPage.vue (hunk 1 above)
Comment (Molunerfinn) [reply src/renderer/pages/UploaderConfigPage.vue:18]: Somethine like this:

image

File: src/renderer/pages/UploaderConfigPage.vue (hunk 2 above)
Comment (Molunerfinn) [reply src/renderer/pages/UploaderConfigPage.vue:25]: Somethine like this:

image

Comment (STDSuperman) [reply src/renderer/pages/UploaderConfigPage.vue:25]: This is still under development.

File: src/renderer/pages/UploaderConfigPage.vue [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    </el-row>
+  </div>
+</template>
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator'
+import dayjs from 'dayjs'
+import { completeUploaderMetaConfig } from '../utils/uploader'
+
+@Component({
+  name: 'OssConfigList'
```
Comment (Molunerfinn) [inline src/renderer/pages/UploaderConfigPage.vue:51]: change this name

Comment (Molunerfinn) [reply src/renderer/pages/UploaderConfigPage.vue:51]: & should add mixin for `defaultPicBed` just like `picbeds/index.vue` do

Comment (Molunerfinn) [reply src/renderer/pages/UploaderConfigPage.vue:51]: image

or will cause this error:

image

File: src/renderer/pages/picbeds/index.vue [hunk 5]
```diff
@@ -58,15 +54,36 @@ export default class extends Vue {
...
+      const configList = await this.getConfig<IStringKeyMap[]>(configListConfigPath)
+      // Finds the specified item from the config array and modifies it
+      const existItem = configList?.find(item => item._id === result._id)
+      // edit
+      if (existItem) {
+        Object.assign(existItem, result, {
+          _updatedAt: Date.now()
+        })
+      } else { // add new
+        configList?.push(completeUploaderMetaConfig(result))
```
Comment (Molunerfinn) [inline src/renderer/pages/picbeds/index.vue:67]: should also add `trimValues` for result
image

File: src/renderer/pages/picbeds/index.vue [hunk 6]
```diff
@@ -58,15 +54,36 @@ export default class extends Vue {
...
-      this.saveConfig({
-        [`picBed.${this.type}`]: trimValues(result)
-      })
+      const configListConfigPath = `uploader.${this.type}.configList`
+      const configList = await this.getConfig<IStringKeyMap[]>(configListConfigPath)
+      // Finds the specified item from the config array and modifies it
+      const existItem = configList?.find(item => item._id === result._id)
+      // edit
+      if (existItem) {
+        Object.assign(existItem, result, {
```
Comment (Molunerfinn) [inline src/renderer/pages/picbeds/index.vue:63]: should also add `trimValues` for result.

**Review by Molunerfinn — CHANGES_REQUESTED** [review]

**Review by Molunerfinn — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**